libs/wss/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+use std::io::{BufReader, Read, Seek, Write};
+
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+
+mod compression;
+mod error;
+mod wav;
+
+pub use compression::Compression;
+pub use error::Error;
+
+pub struct Wss {
+    compression: Compression,
+    format: u16,
+    channels: u16,
+    sample_rate: u32,
+    bytes_per_second: u32,
+    block_align: u16,
+    bits_per_second: u16,
+    output_size: u16,
+    data: Vec<u8>,
+}
+
+impl Wss {
+    /// Read a WSS file from the input
+    ///
+    /// # Errors
+    /// [`std::io::Error`] if an IO error occurs
+    /// [`Error::UnsupportedFileType`] if the file is not a WSS file
+    /// [`Error::InvalidCompressionValue`] if the compression value is invalid
+    pub fn read<I: Read + Seek>(input: &mut I) -> Result<Self, Error> {
+        let mut reader = BufReader::new(input);
+
+        let mut buffer = [0; 4];
+        reader.read_exact(&mut buffer)?;
+        if &buffer != b"WSS0" {
```

Comment:
missmatch means the WSS header (8 bytes) is not present, the file does not use compression (compression=0) and the first byte is the format of the wave header